lib/diff/lcs/ldiff.rb

```diff
@@ -17,6 +17,12 @@ module Diff::LCS::Ldiff # :nodoc:
   MIT licence.
   COPYRIGHT
   # standard:enable Layout/HeredocIndentation
+
+  InputInfo = Struct.new(:filename, :data, :stat) do
+    def initialize(filename)
+      super(filename, ::File.read(filename), ::File.stat(filename))
+    end
+  end
 end
 
 class << Diff::LCS::Ldiff
```

Comment:
Note that this suggestion will cause a bit of a problem, because I cannot extend the change down to line 33, but this replaces everything from line 26 to 33.

```suggestion

  class << self
    attr_reader :format, :lines # :nodoc:
    attr_reader :file_old, :file_new # :nodoc:
    attr_reader :data_old, :data_new # :nodoc:
  end

  def self.run(args, _input = $stdin, output = $stdout, error = $stderr) # :nodoc:
```

Reply:
I'm applying it manually :)

Reply:
As part of your future Major version, don't you think you could remove these attr_readers? I don't see much use for them.

Reply:
I don't remember why they were used, but when it moves to an instance, we will probably replace them with `attr_accessor` so that someone can create an `Ldiff` instance, change settings on it, and then produce the output.

But that is part of what requires thinking: how much should be set up at the beginning vs. how much should be parameters to the output production?